Files/DataModels/SidebarPinnedModel.cs

```diff
@@ -266,20 +267,26 @@ public async Task AddItemToSidebarAsync(string path)
 
                     if (thumbnail != null)
                     {
-                        locationItem.IconData = await thumbnail.ToByteArrayAsync();
-                        locationItem.Icon = await locationItem.IconData.ToBitmapAsync();
+                        await CoreApplication.MainView.DispatcherQueue.EnqueueAsync(async () =>
+                        {
+                            locationItem.IconData = await thumbnail.ToByteArrayAsync();
```

Comment:
I think usually the IconData assignment it's put outside EnqueueAsync as it does not require to be executed on the UI thread.